### PR TITLE
[#ICC-75]  Choose auth method for Third Party Services calls

### PR DESCRIPTION
**Pull request by gquadrati**
This PR depends in #899 

#### List of Changes

- Handle authorization based on Third Party configuration

#### Motivation and Context

Enable Third Party messages on IO

#### How Has This Been Tested?

Unit Tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: openapi/api-third-party.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+swagger: '2.0'
+info:
+  title: Third Party Messages - Server implementation
+  version: '1'
+basePath: /
+schemes:
+  - https
+paths:
+  '/{fiscal_code}/messages/{id}':
```
Comment (fabriziopapi) [inline openapi/api-third-party.yaml:9]: I'm not sure it's safe to use the fiscal_code in path. Maybe we have to move it in the header?

Comment (fabriziopapi) [reply openapi/api-third-party.yaml:9]: @AleDore

File: src/clients/third-party-service-client.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  createClient
+} from "../../generated/third-party-service/client";
+
+/**
+ *
+ * @param apiKey
+ * @returns
+ */
+const withApiKey = (apiKey: ApiKeyAuthenticationConfig) => (
+  fetchApi: typeof fetch = (nodeFetch as unknown) as typeof fetch // TODO: customize fetch with timeout
```
Comment (fabriziopapi) [inline src/clients/third-party-service-client.ts:18]: **NB:** the timeout could be set outside using the AbortableFetch (eg config.ts -> getHttpsApiFetchWithBearer)

Comment (gquadrati) [reply src/clients/third-party-service-client.ts:18]: Sorry, I just copy-paste a cast without removing the comments :D

**Comment by gquadrati** [comment]
Since this PR needs ThirdParty specs to be merged, and since they are out of scope for this PR, I will move it back to Draft 
@AleDore @fabriziopapi

**Pull request merged**